panther_driver/src/panther_hardware.py

```diff
@@ -0,0 +1,212 @@
+        self.charger_enable_service = rospy.Service('/panther_hardware/charger_enable', SetBool, self.handle_charger_enable)
+        self.disable_digital_service = rospy.Service('/panther_hardware/disable_digital_power', SetBool, self.handle_disable_digital_power)
+        self.motors_enable_service = rospy.Service('/panther_hardware/motors_enable', SetBool, self.handle_motors_enable)
+        self.fan_enable_service = rospy.Service('/panther_hardware/fan_enable', SetBool, self.handle_fan_enable)
+        self.reset_e_stop_service = rospy.Service('/panther_hardware/reset_e_stop', Trigger, self.handle_reset_e_stop)
+        self.trigger_e_stop_service = rospy.Service('/panther_hardware/trigger_e_stop', Trigger, self.handle_trigger_e_stop)
+
+        # ROS Publishers with timers
+        self.e_stop_state_pub = rospy.Publisher('/panther_hardware/e_stop', Bool, queue_size=1)
+        self.timer_e_stop = rospy.Timer(rospy.Duration(0.1), self.publish_e_stop_state)
+
+        self.charger_state_pub = rospy.Publisher('/panther_hardware/charger_sens', Bool, queue_size=1)
+        self.timer_charger = rospy.Timer(rospy.Duration(0.5), self.publish_charger_state)
+
+        self.first_start_sequence()
+
+        rospy.spin()
+
+    def first_start_sequence(self):
+        """
+        First start sequence for motors which is meant to power up modules in correct order
+        """
+        GPIO.output(VMOT_ON, 1)
+        time.sleep(0.5)
+
+        GPIO.output(DRIVER_EN, 1)
+        time.sleep(0.2)
+
+        GPIO.output(AUX_PW_EN, 1)
+
+
+    def publish_e_stop_state(self, event=None):
+        msg = Bool()
+        msg.data = self.read_e_stop_pin()
+        self.e_stop_state_pub.publish(msg)
+
+    def publish_charger_state(self, event=None):
+        msg = Bool()
+        msg.data = GPIO.input(CHRG_SENSE)
+        self.charger_state_pub.publish(msg)
+
+    def handle_aux_power_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, AUX_PW_EN, "Aux power enable")
+
+    def handle_charger_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, CHRG_EN, "Charger enable")
+
+    def handle_disable_digital_power(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, VDIG_OFF, "Digital power disable")
+
+    def handle_motors_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, DRIVER_EN, "Motors driver enable")
+
+    def handle_fan_enable(self, req: SetBoolRequest):
+        return self.handle_set_bool_srv(req.data, FAN_SW, "Fan enable")
+
+    
+    def handle_trigger_e_stop(self, req: TriggerRequest):
+        self.watchdog.turn_off()
+        return TriggerResponse(True, f"E-STOP triggered, watchdog turned off")
+
+    def handle_reset_e_stop(self, req: TriggerRequest):
+        # Read value before reset
+        e_stop_initial_val = self.read_e_stop_pin()
+
+        # Check if E-STOP reset is needed
+        print(e_stop_initial_val)
```

Comment:
rospy.logdebug()

Reply:
Will remove the prints altogether as they are redundant with service massages